### PR TITLE
Add pod annotations

### LINKED ISSUE #1848: Helm setting the annotation of controller and allocator
**Is your feature request related to a problem? Please describe.**

I want to set the annotation of the controller/allocator pod.
This is because we need to add values to annotation when we get metrics in datadog.

```yaml
# (...)
metadata:
    #(...)
    annotations:
        ad.datadoghq.com/<CONTAINER_IDENTIFIER>.check_names: |
            ["openmetrics"]
        ad.datadoghq.com/<CONTAINER_IDENTIFIER>.init_configs: |
            [{}]
        ad.datadoghq.com/<CONTAINER_IDENTIFIER>.instances: |
            [
              {
                "prometheus_url": "http://%%host%%:%%port%%/<PROMETHEUS_ENDPOINT> ",
                "namespace": "<METRICS_NAMESPACE_PREFIX_FOR_DATADOG>",
                "metrics": [{"<METRIC_TO_FETCH>":"<NEW_METRIC_NAME>"}]
              }
            ]
spec:
    containers:
        - name: '<CONTAINER_IDENTIFIER>'
```

refs: https://docs.datadoghq.com/agent/kubernetes/prometheus/

**Describe the solution you'd like**

I would like to be able to inject annotations in values.yaml

**Describe alternatives you've considered**

Download the chart and rewrite it in parts.
However, this is less maintainable in the long run.

**Additional context**

None

### PR DESCRIPTION
**Pull request by 8398a7**
**What type of PR is this?**

/kind feature

**What this PR does / Why we need it**:

refs: #1848

**Which issue(s) this PR fixes**:

Closes #1848

**Special notes for your reviewer**:

### CODE REVIEW

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 1]
```diff
@@ -186,6 +186,7 @@ The following tables lists the configurable parameters of the Agones chart and t
 | `agones.controller.nodeSelector`                    | Controller [node labels][nodeSelector] for pod assignment                                       | `{}`                   |
 | `agones.controller.tolerations`                     | Controller [toleration][toleration] labels for pod assignment                                   | `[]`                   |
 | `agones.controller.affinity`                        | Controller [affinity][affinity] settings for pod assignment                                     | `{}`                   |
+| `agones.controller.annotations`                     | Controller [annotations][annotations] settings for pod assignment                               | `{}`                   |
```
Comment (markmandel) [inline site/content/en/docs/Installation/Install Agones/helm.md:189]: Please move this down to where we have the section "New Configuration Features:" - so they only get displayed on the next release :+1: 

Context: https://agones.dev/site/docs/contribute/#within-a-page

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 2]
```diff
@@ -226,6 +227,7 @@ The following tables lists the configurable parameters of the Agones chart and t
 | `agones.allocator.disableTLS`                       | Turns off TLS security for incoming connections to the allocator. | `false`                |
 | `agones.allocator.tolerations`                      | Allocator [toleration][toleration] labels for pod assignment                                    | `[]`                   |
 | `agones.allocator.affinity`                         | Allocator [affinity][affinity] settings for pod assignment                                      | `{}`                   |
+| `agones.allocator.annotations`                      | Allocator [annotations][annotations] settings for pod assignment                                | `{}`                   |
```
Comment (markmandel) [inline site/content/en/docs/Installation/Install Agones/helm.md:230]: Please also move this down to where we have the section "New Configuration Features:" - so they only get displayed on the next release :+1:

**Review by markmandel — COMMENTED** [review]
Looks great! Just one small tweak in the docs.

Not sure if you want to - but you could also do the Ping service while you are in there?
https://github.com/googleforgames/agones/blob/master/install/helm/agones/templates/ping.yaml

**Comment by 8398a7** [comment]
> Looks great! Just one small tweak in the docs.
> 
> Not sure if you want to - but you could also do the Ping service while you are in there?
> https://github.com/googleforgames/agones/blob/master/install/helm/agones/templates/ping.yaml

Do you mean `agones.ping.http.annotations` and `agones.ping.udp.annotations` that already exist?
I had noticed that these weren't documented, but the background is somewhat different, so I thought I'd make a different PR.
But it looks like I can change them together, so I'll change it.

Only ping Deployment doesn't have annotations, which seems strange, so I'll add them together.

File: site/content/en/docs/Installation/Install Agones/helm.md (hunk 1 above)
Comment (8398a7) [reply site/content/en/docs/Installation/Install Agones/helm.md]: Thank you. I fixed it with ab93fd6b5e51ebf65ce76e132a068b1e5584ff2e.

File: site/content/en/docs/Installation/Install Agones/helm.md (hunk 2 above)
Comment (8398a7) [reply site/content/en/docs/Installation/Install Agones/helm.md]: Thank you. I fixed it with ab93fd6b5e51ebf65ce76e132a068b1e5584ff2e.

**Comment by markmandel** [comment]
Not quite seeing it, but there something going wrong with the annotation link:

```
htmltest started at 03:18:20 on /tmp/website
========================================================================
site/docs/installation/install-agones/helm/index.html
  target does not exist --- site/docs/installation/install-agones/helm/index.html --> annotations
========================================================================
✘✘✘ failed in 375.927952ms
1 errors in 118 documents
Html Test: Attempt 2
htmltest started at 03:19:21 on /tmp/website
```

**Comment by 8398a7** [comment]
Oh, moving annotations to "New Configuration Features" seems to have affected it.
I should have moved the annotations link along with it.

I'm trying the `make site-server` at hand, so I'll push it if it's OK.

**Comment by 8398a7** [comment]
It's still no good...
The `make site-server` seems to work, but...

I've written a patch to run the `site-test` at hand and will push it when the test passes.

```diff
diff --git a/build/Makefile b/build/Makefile
index 99cb5843..aa7bb85d 100644
--- a/build/Makefile
+++ b/build/Makefile
@@ -235,6 +235,8 @@ push-release-chart: $(ensure-build-image)
 # Run all tests
 test: $(ensure-build-image) test-go test-sdks test-install-yaml site-test

+site-test: $(ensure-build-image) site-test
+
 # Run go tests
 test-go: $(ensure-build-image)
        $(GO_TEST) $(agones_package)/pkg/... \
```

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 3]
```diff
@@ -236,7 +236,14 @@ The following tables lists the configurable parameters of the Agones chart and t
...
 | --------------------------------------------------- | ----------------------------------------------------------------------------------------------- | ---------------------- |
-|                                                     |                                                                                                 |                        |             
+| `agones.controller.annotations`                     | Controller [annotations][annotations] settings for pod assignment                               | `{}`                   |
+| `agones.allocator.annotations`                      | Allocator [annotations][annotations] settings for pod assignment                                | `{}`                   |
+| `agones.ping.annotations`                           | Ping [annotations][annotations] settings for pod assignment                                     | `{}`                   |
+| `agones.ping.http.annotations`                      | [Annotations](annotations) for ping http service                                                | `{}`                   |
```
Comment (8398a7) [inline site/content/en/docs/Installation/Install Agones/helm.md:242]: `[Annotations](annotations)` 😭

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 4]
```diff
@@ -236,7 +236,14 @@ The following tables lists the configurable parameters of the Agones chart and t
...
 | --------------------------------------------------- | ----------------------------------------------------------------------------------------------- | ---------------------- |
-|                                                     |                                                                                                 |                        |             
+| `agones.controller.annotations`                     | Controller [annotations][annotations] settings for pod assignment                               | `{}`                   |
+| `agones.allocator.annotations`                      | Allocator [annotations][annotations] settings for pod assignment                                | `{}`                   |
+| `agones.ping.annotations`                           | Ping [annotations][annotations] settings for pod assignment                                     | `{}`                   |
+| `agones.ping.http.annotations`                      | [Annotations][annotations] for ping http service                                                | `{}`                   |
```
Comment (markmandel) [inline site/content/en/docs/Installation/Install Agones/helm.md:242]: Sorry to make you juggle this around, but `agones.ping.udp.annotations` and `agones.ping.http.annotations` already exist in the current release, but aren't documented - so they can go in the section up above.

There new, remaining values should stay down here.

Random thought - do we think there is any confusion given that `agones.ping.udp.annotations` is a annotation on a service, but `agones.controller.annotations` is on Pods?

Comment (8398a7) [reply site/content/en/docs/Installation/Install Agones/helm.md:242]: > Random thought - do we think there is any confusion given that agones.ping.udp.annotations is a annotation on a service, but agones.controller.annotations is on Pods?

Hmmm, it's somewhat confusing.
Should I revise it to `agones.controller.podAnnotations` or `agones.ping.udp.serviceAnnotations` or something?

Comment (8398a7) [reply site/content/en/docs/Installation/Install Agones/helm.md:242]: > Sorry to make you juggle this around, but agones.ping.udp.annotations and agones.ping.http.annotations already exist in the current release, but aren't documented - so they can go in the section up above.
>
> There new, remaining values should stay down here.

I understand.
I'll fix the name I asked for in my comment above when I decide what to do with it.

Comment (markmandel) [reply site/content/en/docs/Installation/Install Agones/helm.md:242]: @roberthbailey @pooneh-m would love a second opinion?

Worth noting, `agones.ping.udp.annotations` and `agones.ping.http.annotations` where in previous releases, but weren't documented.

WDYT?

Comment (roberthbailey) [reply site/content/en/docs/Installation/Install Agones/helm.md:242]: If you look at the structure in https://github.com/googleforgames/agones/blob/master/install/helm/agones/values.yaml I don't actually think it's all that confusing. The annotations for the service / pod are clearly nested under the values for the service / pod respectively. 

I think the descriptions in this table should make it clear what type of k8s resource the annotations are being applied to. e.g.

| `agones.controller.annotations` | [Annotations][annotations] added to the Agones controller pods |
| `agones.ping.http.annotations ` | [Annotations][annotations] added to the Agones ping http service |

Comment (markmandel) [reply site/content/en/docs/Installation/Install Agones/helm.md:242]: SGTM! Thanks for the second/third pair of eyes - that works for me :+1:

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 5]
```diff
@@ -236,7 +236,14 @@ The following tables lists the configurable parameters of the Agones chart and t
 
 | Parameter                                           | Description                                                                                     | Default                |
 | --------------------------------------------------- | ----------------------------------------------------------------------------------------------- | ---------------------- |
-|                                                     |                                                                                                 |                        |             
+| `agones.controller.annotations`                     | Controller [annotations][annotations] settings for pod assignment                               | `{}`                   |
```
Comment (markmandel) [inline site/content/en/docs/Installation/Install Agones/helm.md:239]:
```suggestion
| `agones.controller.annotations`                     | [Annotations][annotations] added to the Agones controller pods                               | `{}`                   |
```

Just going by Robbie's suggestion, to make it super clear. WDYT?

File: site/content/en/docs/Installation/Install Agones/helm.md (hunk 4 above)
Comment (markmandel) [inline site/content/en/docs/Installation/Install Agones/helm.md:242]:
```suggestion
| `agones.ping.http.annotations`                      | [Annotations][annotations] added to the Agones ping http service                                             | `{}`                   |
```

Minor tweaks for just some extra clarity based on Robbie's suggestions. Just to make things extra clear.

Comment (8398a7) [reply site/content/en/docs/Installation/Install Agones/helm.md]: Fixed.

File: site/content/en/docs/Installation/Install Agones/helm.md (hunk 5 above)
Comment (8398a7) [reply site/content/en/docs/Installation/Install Agones/helm.md]: Fixed.

File: site/content/en/docs/Installation/Install Agones/helm.md (hunk 4 above)
Comment (8398a7) [reply site/content/en/docs/Installation/Install Agones/helm.md]: Fixed :+1:

**Review by markmandel — APPROVED** [review]
Let's get this in!

**Pull request merged**